src/generic-container.test.ts

```diff
@@ -104,4 +104,59 @@ describe("GenericContainer", () => {
     await startedContainer.stop();
     await expect(fetch(url)).rejects.toThrowError();
   });
+
+  it("should build and start from a Dockerfile with build arguments", async () => {
+    const context = path.resolve(__dirname, "..", "docker-with-buildargs");
+    const container = await GenericContainer.fromDockerfile(context)
+      .withBuildArg("VERSION", "10-alpine")
+      .build();
+    const startedContainer = await container.withExposedPorts(8080).start();
+
+    const url = `http://${startedContainer.getContainerIpAddress()}:${startedContainer.getMappedPort(8080)}`;
+    const response = await fetch(`${url}/hello-world`);
+    expect(response.status).toBe(200);
+
+    await startedContainer.stop();
+    await expect(fetch(url)).rejects.toThrowError();
+  });
+
+  it("should not build and start from a Dockerfile with missing build arguments", async () => {
+    const context = path.resolve(__dirname, "..", "docker-with-buildargs");
+
+    await expect(GenericContainer.fromDockerfile(context).build()).rejects.toThrowError("Failed to build image");
+  });
+
+  it("should work for mysql", async () => {
```

Comment:
These tests have been removed in master, not sure if there was an issue with the merge

Reply:
Why have they been removed? They're actually working great :)

Reply:
They were added when we had reported issues with mysql and couchdb to reproduce. Eventually we found the issue was with port padding in the host port check and a test was added specifically for that. These tests no longer add value and are expensive as the test will likely have to pull these images. We're trying to only use a single image in the build for performance reasons

Reply:
Okay I removed the tests from master ;)